src/wallet/wallet.cpp

```diff
@@ -633,7 +639,9 @@ bool CWallet::EncryptWallet(const SecureString& strWalletPassphrase)
         if (IsHDEnabled()) {
             CKey key;
             CPubKey masterPubKey = GenerateNewHDMasterKey();
-            if (!SetHDMasterKey(masterPubKey))
+            // preserve the old chains version to not break backward compatibility
+            CHDChain oldChain = GetHDChain();
+            if (!SetHDMasterKey(masterPubKey, &oldChain))
```

Comment:
I though hiding the internals at this point would be preferable. IMO `CWallet::encryptWallet` should not have direct knowhow of the CHDChain version handling. I'd prefer passing in the old chain and let it be handled through `SetHDMasterKey`. Also, in future maybe other attributes need to be preserved.
Not sure although.